src/parser.rs

```diff
@@ -637,13 +666,82 @@ fn definition() -> Result<()> {
                     "add".to_string(),
                     simple_procedure(
                         vec!["x".to_string(), "y".to_string()],
-                        Expression::ProcedureCall(
+                        Expression::ProcedureCall(ProcedureCall::new(
                             Box::new(Expression::Identifier("+".to_string())),
                             vec![
                                 Expression::Identifier("x".to_string()),
                                 Expression::Identifier("y".to_string()),
                             ]
-                        )
+                        ))
+                    )
+                ))
+            )
+        }
+        {
+            let tokens = vec![
```

Comment:
This test is not very unit.
Should test eval_tail_expression

For complex test, the regression has already done for you.

Reply:
Ok, I will add some tests for eval_tail_expression